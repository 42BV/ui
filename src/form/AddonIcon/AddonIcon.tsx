import { Icon, IconType } from '../../core/Icon';

import { InputGroupText } from 'reactstrap';

export type Props = {
  icon: IconType;

  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;
};

/**
 * Defines an addon to use with an Input element which is a readonly
 * Icon.
 */
export function AddonIcon({ icon, className }: Props) {
  return (
    <InputGroupText>
      <Icon icon={icon} className={className} />
    </InputGroupText>
  );
}
