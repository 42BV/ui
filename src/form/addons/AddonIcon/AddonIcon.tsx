import React from 'react';
import { Icon, IconType } from '../../..';

import { Addon, Props as AddonProps } from '../Addon/Addon';
import { InputGroupText } from 'reactstrap';

export type Props = Omit<AddonProps, 'children'> & {
  icon: IconType;

  /**
   * Optional extra CSS class you want to add to the component.
   * Useful for styling the component.
   */
  className?: string;
};

/**
 * Defines an addon to use with an Input element which is a readonly
 * Icon.
 */
export function AddonIcon({ icon, className, position }: Props) {
  return (
    <Addon position={position}>
      <InputGroupText>
        <Icon icon={icon} className={className} />
      </InputGroupText>
    </Addon>
  );
}
