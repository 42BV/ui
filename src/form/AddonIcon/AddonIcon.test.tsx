import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { AddonIcon } from './AddonIcon';

describe('Component: AddonIcon', () => {
  function setup({ className }: { className?: string }) {
    const { container } = render(
      <AddonIcon icon="360" className={className} />
    );

    return { container };
  }

  test('ui', () => {
    const { container } = setup({});
    expect(container).toMatchSnapshot();
  });

  test('custom class', () => {
    setup({ className: 'extra-class' });
    expect(screen.getByText('360')).toHaveClass('extra-class');
  });
});
