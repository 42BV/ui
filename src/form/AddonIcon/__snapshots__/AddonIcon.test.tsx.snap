// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AddonIcon ui 1`] = `
<div>
  <span
    class="input-group-text"
  >
    <i
      class="icon material-icons material-icons__filled"
    >
      360
    </i>
  </span>
</div>
`;
